parlai/tasks/wikipedia/agents.py

```diff
@@ -76,6 +78,92 @@ def get_extraction_instructions(self):
         return instructions
 
 
+class FullSplitTeacher(ChunkTeacher):
+    """
+    Full Wikipedia teacher that splits the chunks into train/valid/test.
+    """
+
+    def __init__(self, opt, shared=None):
+        if shared is None:
+            # set map
+            self.opt = opt
+            self._set_chunk_idx_to_file()
+        else:
+            self.chunk_idx_to_file = shared['chunk_idx_to_file']
+        super().__init__(opt, shared)
+
+    def _get_data_folder(self):
+        return os.path.join(self.opt['datapath'], 'wikipedia/full/wiki_full_extracted')
+
+    def get_num_samples(self, datatype) -> int:
+        """
+        Return the number of samples given the datatype.
+        """
+        if 'train' in datatype:
+            return 5437097, 5437097
+        elif 'valid' in datatype:
+            return 71052, 71052
+        else:
+            # test
+            return 39975, 39975
+
+    def _set_chunk_idx_to_file(self):
+        folder = self._get_data_folder()
+        all_subdirs = sorted([x for x in os.listdir(folder) if 'README' not in x])
+        self.chunk_idx_to_file = {i: x for i, x in enumerate(all_subdirs)}
+
+    def get_fold_chunks(self, datatype) -> List[int]:  # type: ignore
+        """
+        Return a list of chunk IDs (integer).
+
+        Given the datatype (train/test/valid), return the list of chunk IDs that
+        correspond to that split.
+        """
+        all_subdirs = sorted([x for x in os.listdir(self.folder) if 'README' not in x])
```

Comment:
it looks like this func is called on every `__init__`, perhaps we could cache this `all_subdirs` somewhere so we don't have to sort it every time? not necessary if this is not a lot of subdirs, just didn't know the number

Reply:
oh yes... i made a mistake here. i half-fixed this problem and did not push the other half lol